### PR TITLE
Small fixes to web UI

### PR DESCRIPTION
**Pull request by mateiz**
- Use SPARK_PUBLIC_DNS environment variable if set (for EC2); this fixes SPARK-833 (https://spark-project.atlassian.net/browse/SPARK-833)
- Use a non-ephemeral port (3030 instead of 33000) by default
- Updated test to use non-ephemeral port too

### CODE REVIEW

File: core/src/test/scala/spark/ui/UISuite.scala [hunk 1]
```diff
@@ -24,7 +24,7 @@ import org.eclipse.jetty.server.Server
 
 class UISuite extends FunSuite {
   test("jetty port increases under contention") {
-    val startPort = 33333
+    val startPort = 3030
     val server = new Server(startPort)
     server.start()
     val (jettyServer1, boundPort1) = JettyUtils.startJettyServer("localhost", startPort, Seq())
```
Comment (rxin) [inline core/src/test/scala/spark/ui/UISuite.scala:30]: While you are at this one, maybe we can change the test to just check startPort + delta > boundPort1 > startPort, and boundPort1 + delta > boundPort2 > boundPort1? 

Setting delta to 10 would get rid of the occasional jenkins failures.

Comment (mateiz) [inline core/src/test/scala/spark/ui/UISuite.scala:30]: Yeah good idea

**Comment by mateiz** [comment]
Jenkins, please retest this

File: ec2/spark_ec2.py [hunk 2]
```diff
@@ -178,7 +178,7 @@ def launch_cluster(conn, opts, cluster_name):
     master_group.authorize('tcp', 50030, 50030, '0.0.0.0/0')
     master_group.authorize('tcp', 50070, 50070, '0.0.0.0/0')
     master_group.authorize('tcp', 60070, 60070, '0.0.0.0/0')
-    master_group.authorize('tcp', 33000, 33010, '0.0.0.0/0')
+    master_group.authorize('tcp', 3030, 3030, '0.0.0.0/0')
```
Comment (shivaram) [inline ec2/spark_ec2.py]: On a similar note, allow some wiggle room here in case you want to launch two `spark-shell` at the same time on EC2 ?

Comment (mateiz) [inline ec2/spark_ec2.py]: OK, done.

**Comment by mateiz** [comment]
Merging this since it passed the tests; thanks for the feedback.

**Comment by pwendell** [comment]
@mateiz  - Could you also update `docs/configuration.md` which currently lists the default as 33000?

**Comment by mateiz** [comment]
Good catch; done.

**Pull request merged**